litr/src/main/java/com/linkedin/android/litr/render/Renderer.java

```diff
@@ -32,12 +32,15 @@
     @Nullable Surface getInputSurface();
 
     /**
-     * Render a frame
-     * @param frame {@link Frame} to operate with. Non-null ror non OpenGL renderer, will contain raw pixels.
+     * Render an input frame
+     * @param inputFrame {@link Frame} to operate with. Non-null ror non OpenGL renderer, will contain raw pixels.
      *                           null for GL renderer, which should assume that environment has been set and just invoke Gl calls.
      * @param presentationTimeNs frame presentation time in nanoseconds
+     * @param sourceMediaFormat the raw(source) {@link MediaFormat}
+     * @param targetMediaFormat the target {@link MediaFormat}
      */
-    void renderFrame(@Nullable Frame frame, long presentationTimeNs);
+    void renderFrame(@Nullable Frame inputFrame, long presentationTimeNs, MediaFormat sourceMediaFormat,
```

Comment:
Why do we have to break a public API? Source and target formats are passed in `init` method, can't we just hold on to them in the renderer? They don't usually change during transcoding.

Reply:
No, we don't need to break the `renderFrame()` method, but yes, they change after `init` method, so I'll add a new method `onMediaFormatChanged()` to `Renderer` interface, and will call it each time the `MediaCodec.INFO_OUTPUT_FORMAT_CHANGED` is returned